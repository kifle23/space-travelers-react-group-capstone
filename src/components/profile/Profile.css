.profile-content {
  display: grid;
  grid-template-columns: repeat(2, minmax(0, 1fr));
  grid-template-rows: auto;
  gap: 20px;
  width: 100%;
  max-width: 1024px;
  margin-bottom: 60px;
}

.mission-list {
  border: 1px solid hsl(180, 87%, 34%);
  display: grid;
  grid-template-rows: repeat(auto-fit, minmax(0, 1fr));
  gap: 1px;
}

.user-missions {
  list-style: none;
  padding: 0;
  margin: 0;
}

.user-missions li {
  display: grid;
  grid-template-columns: 1fr 2fr;
  align-items: center;
  border-bottom: 1px solid hsl(180, 87%, 34%);
  padding: 20px;
}

.action-btns .leave-mission-btn {
  background-color: #f00;
  color: #fff;
  border: 1px solid #f00;
  padding: 5px;
  border-radius: 5px;
  cursor: pointer;
}

.action-btns .read-more-btn,
.action-btns .join-btn {
  background-color: transparent;
  border: 1px solid #6c757e;
  padding: 5px;
  border-radius: 5px;
  color: #6c757e;
  cursor: pointer;
}

.action-btns {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  gap: 5px;
}

.mission-name {
  grid-column: 1 / span 1;
}
