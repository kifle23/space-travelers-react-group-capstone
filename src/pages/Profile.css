.profile-section {
  display: flex;
  justify-content: center;
  height: 100%;
  align-content: center;
  margin-bottom: 60px;
  border-bottom: 1px solid #e7e7e7;
  flex-direction: column;
  align-items: center;
}
