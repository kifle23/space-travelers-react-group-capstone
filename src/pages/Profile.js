const Profile = () => (
  <div className="container"><h2>Profile</h2></div>
);

export default Profile;
