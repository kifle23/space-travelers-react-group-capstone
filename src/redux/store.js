import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './features/Rockets/RocketSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
  },
});

const selectRockets = (store) => store.rockets;
export { selectRockets };
export default store;
