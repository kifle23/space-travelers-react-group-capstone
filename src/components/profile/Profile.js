import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { updateMission } from '../../redux/missionSlice';
import './Profile.css';

const Profile = ({ missions }) => {
  const dispatch = useDispatch();

  const handleMission = (event, id) => {
    dispatch(updateMission(id));
  };

  const readMore = (event, url) => {
    window.open(url, '_blank', 'noopener,noreferrer');
  };

  return (
    <div className="profile-content">
      <div className="missions">
        <h1>Missions</h1>
        <div className="mission-list">
          {missions.length ? (
            <ul className="user-missions">
              {missions.map(({ id, missionName, url }) => (
                <li key={id}>
                  <p className="mission-name">{missionName}</p>
                  <div className="action-btns">
                    <button
                      type="button"
                      className="leave-mission-btn"
                      onClick={(event) => handleMission(event, id)}
                    >
                      Leave Mission
                    </button>
                    <button
                      type="button"
                      className="read-more-btn"
                      onClick={(event) => readMore(event, url)}
                    >
                      Read More
                    </button>
                  </div>
                </li>
              ))}
            </ul>
          ) : (
            <p>No Mission Joined</p>
          )}
        </div>
      </div>
    </div>
  );
};

Profile.propTypes = {
  missions: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      missionName: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      url: PropTypes.string.isRequired,
      reserved: PropTypes.bool.isRequired,
    }),
  ).isRequired,
};

export default Profile;
