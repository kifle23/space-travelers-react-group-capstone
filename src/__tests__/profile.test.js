import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/store';
import Profile from '../components/profile/Profile';

it('Test if my_profile renderers correctly', () => {
  const missions = [];
  const renderedMission = renderer.create(
    <Provider store={store}>
      <Profile missions={missions} />
    </Provider>,
  );
  expect(renderedMission).toMatchSnapshot();
});
