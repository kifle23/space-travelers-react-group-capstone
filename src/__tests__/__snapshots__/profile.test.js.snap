// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if my_profile renderers correctly 1`] = `
<div
  className="profile-content"
>
  <div
    className="missions"
  >
    <h1>
      Missions
    </h1>
    <div
      className="mission-list"
    >
      <p>
        No Mission Joined
      </p>
    </div>
  </div>
</div>
`;
